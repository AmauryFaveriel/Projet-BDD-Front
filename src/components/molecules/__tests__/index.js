import React from 'react';
import renderer from 'react-test-renderer';

// This is the only way I found to test our styled components without specific logic

// Import component
import Description from '../description';

describe('Description', () => {
  const spy = jest.fn();
  it('renders correctly', () => {
    const tree = renderer.create(<Description
      bigTitle="Let's cook"
      subTitle="with ottoenghi"
      text="Yotam Ottolenghi—the chef behind instant cookbook classics Jerusalem, Plenty, Ottolenghi Simple and a string of beloved London restaurants—has been at the helm of the culinary vegetable renaissance."
      buttonContent="Discover the recipe"
      testid="description"
      onClick={spy}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
