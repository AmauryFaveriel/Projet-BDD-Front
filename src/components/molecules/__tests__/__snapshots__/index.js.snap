// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description renders correctly 1`] = `
<div
  className="sc-AxhUy gRNaaC"
  data-testid="description"
>
  <label
    className="sc-AxjAm fSUMXK"
  >
    <span>
      Let's cook
    </span>
  </label>
  <label
    className="sc-AxirZ FIgky"
  >
    <span>
      with ottoenghi
    </span>
  </label>
  <label
    className="sc-AxiKw eCeNVD"
  >
    <span>
      Yotam Ottolenghi—the chef behind instant cookbook classics Jerusalem, Plenty, Ottolenghi Simple and a string of beloved London restaurants—has been at the helm of the culinary vegetable renaissance.
    </span>
  </label>
  <button
    className="sc-AxhCb hIDtBL"
    data-testid="button"
    disabled={false}
    onClick={[Function]}
    size="medium"
    width="245px"
  >
    <span>
      Discover the recipes
    </span>
  </button>
</div>
`;
